### PR TITLE
Condense and clarify filename rules, refactor verification

### PR DESCRIPTION
**Pull request by tkfu**
This is a condensing and refactoring of the file naming rules. I believe the meaningful content is identical.

I've also done some minor refactoring of the verification section:

* Removed long explanations of how to construct filenames, replaced with a link to the relevant section
* Moved step 12 of full verification to the proper section--it's about verifying the image itself, not verifying the metadata

**HOWEVER**, as I was doing this, I noticed that there appears to be a contradiction between the full verification and the file naming rules. The file naming rules specifically say that the root metadata should NOT have a version in the filename, while the verification steps try to check for a versioned filename. @JustinCappos / @trishankatdatadog / @awwad: which one is correct?

### CODE REVIEW

**Comment by trishankatdatadog** [comment]
thanks Jon

i'll take a look sometime this weekend, crazy busy at work

**Comment by tkfu** [comment]
@trishankatdatadog confirmed in our call this week that the file naming rule for the root metadata should be that is MUST be versioned. I've now updated this PR to reflect that; one positive review and I'll merge it in.

File: uptane-standard.md [hunk 1]
```diff
@@ -705,9 +704,6 @@ In order to perform full verification, an ECU SHALL perform the following steps:
...
-12. Verify the desired image against its targets metadata.
-  1. If there is no targets metadata about this image, abort the update cycle and report that there is no such image.
-  2. Otherwise, download the image (up to the number of bytes specified in the targets metadata), and verify that its hashes match the targets metadata. (We download up to this number of bytes, because in some cases, the exact number is unknown. This may happen, for example, if an external program is used to compute the root hash of a tree of targets files, and this program does not provide the total size of all of these files.) If consistent snapshots are not used {{metadata_filename_rules}}, then the filename used to download the image file is of the fixed form FILENAME.EXT (e.g., foobar.tar.gz). Otherwise, the filename is of the form HASH.FILENAME.EXT (e.g., c14aeb4ac9f4a8fc0d83d12482b9197452f6adf3eb710e3b1e2b79e8d14cb681.foobar.tar.gz), where HASH is one of the hashes of the targets file listed in the targets metadata file found earlier in step 4. In either case, the client MUST write the file to non-volatile storage as FILENAME.EXT.
```
Comment (JustinCappos) [inline uptane-standard.md]: Why is this removed?

Comment (trishankatdatadog) [reply uptane-standard.md]: Yeah, this is pretty important

Comment (tkfu) [reply uptane-standard.md]: Downloading the image binary and verifying the image against the hashes in the metadata are sections 5.4.3.3. and 5.4.3.4., respectively. This section is explicitly just for *metadata* download and verification.

The rationale for this is that it lets the process be understood as simple, discrete steps:

1. Verify latest attested time (Section 5.4.3.1)
2. Verify metadata (Section 5.4.3.2)
3. Download latest image (Section 5.4.3.3)
4. Verify image (Section 5.4.3.4)
5. Create and send version report (Section 5.4.3.5)

Since metadata verification is so long, involved, and important to Uptane overall, the procedures for partial and full metadata verification are broken out into their own section, and simply referenced from section 5.4.3.2. I could have stuck both procedures as sub-sections of 5.4.3.2 instead, but in my editorial judgement it doesn't flow well that way.

Comment (trishankatdatadog) [reply uptane-standard.md]: Got it. Where do I preview a rendering of this file, please?

**Comment by tkfu** [comment]
I’m on mobile right now, but the link is in one of my first messages above.

On Nov 30, 2018, at 19:04, Trishank K Kuppusamy <notifications@github.com<mailto:notifications@github.com>> wrote:


@trishankatdatadog commented on this pull request.

________________________________

In uptane-standard.md<https://emea01.safelinks.protection.outlook.com/?url=https%3A%2F%2Fgithub.com%2Fuptane%2Fuptane-standard%2Fpull%2F20%23discussion_r237950396&data=01%7C01%7C%7Cdab78a8e39b24c17540f08d656ee389d%7C6d4034cd72254f72b85391feaea64919%7C1&sdata=hnEQJBvWPForAxSwmJ0aj%2BrlRpKnLg12GjRyOO9GbJ4%3D&reserved=0>:

> @@ -705,9 +704,6 @@ In order to perform full verification, an ECU SHALL perform the following steps:
         1. Check that the non-custom metadata (i.e., length and hashes) of the unencrypted image are the same in both sets of metadata.
         2. Check that the custom metadata (e.g., hardware identifier and release counter) are the same in both sets of metadata.
         3. Check that the release counter in the previous targets metadata file is less than or equal to the release counter in this targets metadata file.
-12. Verify the desired image against its targets metadata.
-  1. If there is no targets metadata about this image, abort the update cycle and report that there is no such image.
-  2. Otherwise, download the image (up to the number of bytes specified in the targets metadata), and verify that its hashes match the targets metadata. (We download up to this number of bytes, because in some cases, the exact number is unknown. This may happen, for example, if an external program is used to compute the root hash of a tree of targets files, and this program does not provide the total size of all of these files.) If consistent snapshots are not used {{metadata_filename_rules}}, then the filename used to download the image file is of the fixed form FILENAME.EXT (e.g., foobar.tar.gz). Otherwise, the filename is of the form HASH.FILENAME.EXT (e.g., c14aeb4ac9f4a8fc0d83d12482b9197452f6adf3eb710e3b1e2b79e8d14cb681.foobar.tar.gz), where HASH is one of the hashes of the targets file listed in the targets metadata file found earlier in step 4. In either case, the client MUST write the file to non-volatile storage as FILENAME.EXT.


Got it. Where do I preview a rendering of this file, please?

File: uptane-standard.md (hunk 1 above)
Comment (tkfu) [reply uptane-standard.md]: https://uptane.github.io/uptane-standard/PR-20/uptane-standard.html

**Comment by patrickvacek** [comment]
I think we discussed the issues that had been brought up with this PR during the last call, and my understanding was that everything was resolved. Was there still anything holding this back, or additional concerns that had not been addressed?

**Comment by tkfu** [comment]
I don't think there are any outstanding concerns--I was just waiting for @JustinCappos to change his review to approved. It would be nice to get this in. I'm not sure if I will be able to make it onto today's call (I'll be on a train), but could someone bring this up if it's not merged before then?

**Comment by trishankatdatadog** [comment]
On Fri, Dec 14, 2018 at 6:24 AM OYTIS <notifications@github.com> wrote:

> *@OYTIS* commented on this pull request.
>
> Minor one, but do we really need .EXT here?
>

I think it's fine
